plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -191,6 +191,16 @@ internal fun mapSnippetFindings(
         results += mappedSnippets
     }
 
+    if (runningSnippetCount >= snippetsLimit) {
```

Comment:
Shouldn't this only be `>`? (And the message below say "exceeded" instead of "reached".)

Reply:
Not sure if it's intended, but the message still says "reached" instead of "exceeded".

Reply:
@sschuberth I come back to you on this review because the tests are currently failing, for instance:
https://github.com/bosch-io/oss-review-toolkit/blob/nnobelis/limit_fossid_snippets/plugins/scanners/fossid/src/test/kotlin/FossIdSnippetChoiceTest.kt#L724-L767

This is because the test does the following:
- List the two pending files
- List the snippet of each file with a limit of two snippets.
- Since each pending file has two snippets, the scanner should stop after the first pending file, and output an issue.

So here `runningSnippetCount` is 2 and `snippetsLimit` is also 2, so the comparison was correct.
You can argue that this issue should only be present if the limit is strictly exceeded, but this the above example this is not the case and you need this issue to know there are more snippets results.

Reply:
Indeed, my expectation would be that the issue only gets created if there are more snippets than the limit allows. In other words, if there are exactly as many snippets as the limit allows, there should be no issue filed. Is that the case currently?

Reply:
It is the case currently but I see some problems with that:

- Right now, if we stop exactly at the limit, no issue is produced: what does it means for the user:
  - Is it because there were exactly `<limit>` snippets ?
  - Is it because the limit has been reached (and there are more snippets) ?
- There _must_ be an issue if there a more snippets than what the ORT results currently contains. This issue is present in the snippet report and this is the only way for the user to know that some snippet choices must be done to see the rest of the snippets (See my comment to Frank: https://github.com/oss-review-toolkit/ort/pull/8616#discussion_r1593635819)

We can't peek to see if there are more snippets in the remaining files because of the lazy loading. I mean we could but it would defeat the optimization that the limit allowed.

Reply:
That's why I went with  `>=`: if we stop exactly at the limit, an issue is produced and the user hence knows an another run must be done.

After the second run, if there are more snippets, they are shown in the report. If there are none, the report is empty. Which is a minor annoyance but at least no information is hidden.

Reply:
> I mean we could but it would defeat the optimization that the limit allowed.

Ok, that was the relevant bit of information for me.